### ISSUE TITLE
YOLOE visual prompts (SAVPE): align upstream cpp_ggml runtime with AICore (optrace-level bisection)

### ISSUE THREAD

**Issue by Asher-1** [Asher-1/ACloudViewer#151]
## Summary

Upstream `dl/ultralytics-ggml` (cpp_ggml) visual-prompt (SAVPE) runtime still returns zero detections for most YOLOE checkpoints, while AICore (this repo, `core/AICore`) detects correctly with a **byte-identical vpe**. The remaining divergence is at framework level and needs an optrace-style per-op alignment pass.

## What is already aligned (verified by kept-alive dumps)

| Quantity | AICore vs upstream | Method |
|---|---|---|
| savpe FPN inputs (P3/P4/P5) | cos = 1.0000 | `AICORE_SAVPE_DUMP` vs `YOLO_SAVPE_DUMP` (n-f32 CPU) |
| prompt masks (letterbox rasterization) | rects identical | `[vp-dbg]` / `[savpe-dbg]` rect logs |
| vpe ([512, Q] cls embeddings) | cos = 1.0000, norms equal | kept-alive dumps |
| emb (level-0 embed conv output) | cos = 0.99999, max rel ≈ 2.6% | `YOLO_EMB_DUMP` vs `AICORE_SAVPE_DUMP_emb0` |

## Fixes already synced upstream

1. savpe aggregation `s_g` view: row-offset (`g*nb[0]`, read group 0 at grid g+p, out-of-bounds for g+p>=HW3) → column-offset (`g*nb[1]`).
2. `xT`/`yT` staging permute axes (1,2,0) → (2,0,1) so the flattened grid order matches the mask leaf (h*W+w).
3. `world_detect` e2: `view_2d(..., eT32->nb[1], 0)` → `reshape_2d` (memory reinterpretation of the contiguous [embed,W,H] staging).

Text-prompt regression passes after all three (3x person @0.83, n-f16). Visual: m-f16 now emits 1 detection (was 0); n still 0.

## Suspected root cause of the residual

Upstream cpp_ggml's ggml does **not** carry ACloudViewer's patches (notably `upstream_accuracy/0001-world-f32-gemm-tf32-route-and-im2col-race`). fpn/emb show 2.6%–13% max-element deviations between the two stacks (same GGUF, same input, CPU/F32), which the contrastive head amplifies: upstream max cls sigmoid ≈ 4e-5 (logits ≈ -10 after ls=2.28/bs=-17.16) vs detections on AICore for the same vpe.

## Next steps (optrace-style alignment)

1. Port or reimplement the kept-alive per-op dump (`keep_all_ops`) upstream, then diff every op checksum between the two stacks on the same GGUF/input.
2. Bisect the first divergent op; confirm whether it is im2col accumulation order, the TF32/f32 gemm route, or a patch-specific kernel.
3. Either port the minimal patch(es) upstream or gate the savpe visual-prompt feature on the patched runtime.

## Repro

```bash
# AICore (this repo) — passes 3/3 strict on n/s/l/x, m after the fix
AICORE_TEST_YOLO_SAVPE_GGUF=yoloe-26n-seg-f32.gguf \
AICORE_TEST_YOLO_IMAGE=party_hats.jpg \
AICORE_TEST_YOLO_VP_BOXES="60,95,285,390,940,0,1170,185,455,365,660,650" \
AICORE_TEST_YOLO_SAVPE_DEVICE=cpu ./test_yolo_capi_savpe

# upstream — zero detections
cpp_ggml/build-cpu/bin/yolo-cli detect --model yoloe-26n-seg-f32.gguf \
  --source party_hats.jpg \
  --vp-boxes "60,95,285,390,940,0,1170,185,455,365,660,650" --conf 0.01
```

**Comment by Asher-1** [comment]
already fixed
